### ISSUE TITLE
Implement energy cut-off for Thomas-Fermi model

### ISSUE THREAD

**Issue by n01r** [ComputationalRadiationPhysics/picongpu#1967]
This is a follow-up issue to #1444.

In order to calculate a local "temperature" only from the *bulk* / *thermal* electrons in the plasma an energy cut-off has to be implemented for the `ThomasFermi` impact ionization model. 

- [x] implement the cut-off
- [x] add user documentation

**Comment by n01r** [comment]
I have uploaded my energy cutoff branch to my repo.
https://github.com/n01r/picongpu/tree/topic-AddEnergyCutOffThomasFermi

It currently does not compile and I do not exactly know why. 
I also planned testing it with the `FoilLCT` example, with `laserNone` (no laser), initial electron temperature of 5 keV, one time with a drift in x-direction of 200 keV and one time without. I set the energy cutoff value to 50 keV. I would thus expect more ionization to happen in the first case and none in the second.

Furthermore one can check the energy density with cutoff in the HDF5 data by adding an `EnergyDensityCutoffOperation`  for the species.
